docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/src/product_management_system/vegetable_consumer.bal

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2019 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/kafka;
+import ballerina/log;
+import ballerina/lang.'string as strings;
+import ballerina/io;
+
+// Kafka Consumer Configuration
+kafka:ConsumerConfig vegetableConsumerConfig = {
+    bootstrapServers: "localhost:9092",
+    groupId: "consumer",
+    topics: ["product-price"],
+    pollingIntervalInMillis: 1000,
+    partitionAssignmentStrategy: "org.apache.kafka.clients.consumer.RoundRobinAssignor"
+};
+
+// Kafka Listener
+listener kafka:Consumer vegetableConsumer = new (vegetableConsumerConfig);
+
+// Service that listens to the particular topic
+service vegetableConsumerService on vegetableConsumer {
+    // Trigger whenever a message is added to the subscribed topic
+    resource function onMessage(kafka:Consumer productConsumer, kafka:ConsumerRecord[] records) returns error? {
+        foreach var entry in records {
+            byte[] serializedMessage = entry.value;
+            string|error stringMessage = strings:fromBytes(serializedMessage);
```

Comment:
```suggestion
            string stringMessage = check strings:fromBytes(serializedMessage);
```
Shall we use `check` since we are not handling the error case?
Also shall we remove the if condition after the above chnage?

Reply:
Done